cmd/sealer/cmd/version.go

```diff
@@ -19,31 +19,89 @@ import (
 	"fmt"
 
 	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
 
-	"github.com/sealerio/sealer/version"
+	"github.com/sealerio/sealer/pkg/version"
 )
 
-var shortPrint bool
+var (
+	shortPrint bool
+	output     string
+)
+var sealerErr error
 
 func NewVersionCmd() *cobra.Command {
 	versionCmd := &cobra.Command{
 		Use:     "version",
-		Short:   "show sealer and related versions",
+		Short:   "Print version info",
 		Args:    cobra.NoArgs,
 		Example: `sealer version`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			marshalled, err := json.Marshal(version.Get())
-			if err != nil {
-				return err
+			// Validate validates the provided options.
+			if output != "" && output != "yaml" && output != "json" {
+				return fmt.Errorf("output format must be yaml or json")
 			}
 			if shortPrint {
 				fmt.Println(version.Get().String())
-			} else {
-				fmt.Println(string(marshalled))
+				return nil
 			}
-			return nil
+			return PrintInfo()
 		},
 	}
-	versionCmd.Flags().BoolVar(&shortPrint, "short", false, "if true, print sealer's own version number.")
+	versionCmd.Flags().BoolVar(&shortPrint, "short", false, "If true, print just the version number.")
+	versionCmd.Flags().StringVarP(&output, "output", "o", "yaml", "choose `yaml` or `json` format to print version info")
 	return versionCmd
 }
+
+func PrintInfo() error {
+	OutputInfo := &version.Output{}
+	OutputInfo.SealerVersion = version.Get()
+
+	if err := PrintToStd(OutputInfo); err != nil {
+		return err
+	}
+	//TODO!
+	// missinfo := []string{}
+	// if OutputInfo.KubernetesVersion == nil {
```

Comment:
version just show sealer git info. no need to include cluster info. or we can support it later by using another cmd.